src/ucs/datastruct/bitmap.h

```diff
@@ -126,71 +289,120 @@ _UCS_BITMAP_DECLARE_TYPE(256)
  * @param _bitmap Clear all bits in this bitmap
  */
 #define UCS_BITMAP_CLEAR(_bitmap) \
-    memset((_bitmap)->bits, 0, sizeof((_bitmap)->bits))
+    memset((_bitmap).bits, 0, sizeof((_bitmap).bits))
 
 
 /**
- * Iterate over all set (1) bits of a given bitmap
+ * Find the index of the first bit set to 1 in a given bitmap
  *
- * @param _bitmap Iterate over bits of this bitmap
- * @param _index  Bit index (global offset - relative to the whole bitmap)
+ * @param _bitmap Look for the first bit in this bitmap
  */
-#define UCS_BITMAP_FOR_EACH_BIT(_bitmap, _index) \
-    for (_index = ucs_ffs64_safe((_bitmap)->bits[0]); \
-         _index < ucs_static_array_size((_bitmap)->bits) * UCS_BITMAP_BITS_IN_WORD; \
-         _index = _UCS_BITMAP_WORD_INDEX0(_index) + \
-                  ucs_ffs64_safe(_UCS_BITMAP_WORD((_bitmap), (_index)) & \
-                                 _UCS_BITMAP_GET_NEXT_BIT(_index))) \
-        if (UCS_BITMAP_GET((_bitmap), (_index)))
+#define UCS_BITMAP_FFS(_bitmap) \
+    ({ \
+        size_t _word_index, _index; \
+        _UCS_BITMAP_FOR_EACH_WORD(_bitmap, _word_index) { \
+            _index = ucs_ffs64_safe(_UCS_BITMAP_WORD(_bitmap, _word_index)) + \
+                     _word_index * UCS_BITMAP_BITS_IN_WORD; \
+            if (_index % UCS_BITMAP_BITS_IN_WORD) { \
```

Comment:
how does it work?

Reply:
It returns the same value as `__fs64` would return if it was "extended" to larger words.
Iterate over all 64-bit words, if `ucs_ffs64_safe` returned **not** 64 (the is a "1" bit in **this** word) - return, otherwise continue to the next word.

Reply:
need to check before line 304 if work is not 0 (add gtest)